src/utils/contexts/balances/hook.tsx

```diff
@@ -0,0 +1,44 @@
+import { CurrencyExt } from '@interlay/interbtc-api';
+import { AccountId } from '@polkadot/types/interfaces';
+import { useErrorHandler } from 'react-error-boundary';
+import { useQuery, UseQueryResult } from 'react-query';
+import { useSelector } from 'react-redux';
+
+import { StoreType } from '@/common/types/util.types';
+import { useGetCurrencies } from '@/utils/hooks/api/use-get-currencies';
+import useAccountId from '@/utils/hooks/use-account-id';
+
+import { BalanceData } from './types';
+
+const BALANCES_QUERY_KEY = 'getBalances';
+
+const getBalances = async (currencies: CurrencyExt[], accountId: AccountId): Promise<BalanceData> => {
+  const chainBalances = await Promise.all(
+    currencies.map((currency) => window.bridge.tokens.balance(currency, accountId))
+  );
+
+  return chainBalances.reduce(
+    (acc, balance) => ({
+      ...acc,
+      [balance.currency.ticker]: balance
+    }),
+    {}
+  );
+};
+
+const useGetBalances = (): UseQueryResult<BalanceData | undefined> => {
+  const { bridgeLoaded } = useSelector((state: StoreType) => state.general);
+  const accountId = useAccountId();
+  const { data: currencies, isSuccess } = useGetCurrencies(bridgeLoaded);
```

Comment:
`isSuccess` could be renamed to `isSuccessCurrencies` to have more clarity about what this variable refers to

Reply:
if I change one I would have to change for the rest of the useQuery variables, otherwise we loose consistency

Reply:
@danielsimao I see the `data` field is already renamed here so I don't think we would lose consistency, but get more consistency - all properties obtained from `useGetCurrencies` would be consistently named with `currency` in the name, or am I missing something here?

Reply:
Sorry, once again I misunderstood what you said. Apologies. i will make the changes.